### PR TITLE
Replaced HTTP where HTTPS was available

### PR DESCRIPTION
**Pull request by yackermann**
I found that most of the links in current style guide still been using HTTP. I went through all HTTP links and those that had supported HTTPS been replaces with HTTPS version.

4 links about VIM been using goo.gl link shortening(https://goo.gl/CUXJZC). I have replaced them with actual link.

### CODE REVIEW

**Comment by yackermann** [comment]
I signed it!

2016-04-01 21:48 GMT+13:00 googlebot notifications@github.com:

File: cpplint/README [hunk 1]
```diff
@@ -1,5 +1,5 @@
 This is automated checker to make sure a C++ file follows Google's C++ style
-guide (http://google.github.io/styleguide/cppguide.html). As it
+guide [https://google.github.io/styleguide/cppguide.html](https://google.github.io/styleguide/cppguide.html). As it
```
Comment (geoffromer) [inline cpplint/README]: This doesn't seem to be a Markdown file, so I don't think it should have Markdown-style links.

**Comment by yackermann** [comment]
@geoffromer Ooops. True XD

**Comment by yackermann** [comment]
@geoffromer Done!*)

**Pull request merged**